<a href="https://colab.research.google.com/github/KanayaDzikra/pemograman-lanjutan/blob/main/KanayaDzikra-2C22230015.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**🖥️👩🏻‍💻PRAKTIKUM PEMROGRAMAN LANJUT🔥**
##**Pertemuan (2) PENGULANGAN (LOOP) - Senin, 21 Oktober 2024**

👤 Nama: Kanaya Dzikra Setiadi / NRP: 2C2230015 / Kelas: III/A Sains Data

💬Deskripsi Pembahasan Materi Pengulangan:

1.1.  Pengulangan (LOOP) 🔂
- While Loop 🐋

While dalam phyton bersifat indefinite iteration. Indefinite iteration adalah proses iterasi yang akan berhenti ketika memenuhi kondisi tertentu.
Kondisi output akan berupa TRUE atau FALSE
- For Loop

Struktur ini digunakan bila kita mengetahui secara pasti banyaknya pengulangan yang akan dilakukan. Pernyataan FOR mempunyai 3 parameter yaitu :

-Nilai awal (Start)

-Test kondisi yang menentukan akhir loop (Stop)

-penentu perubahan nilai (Step)


1.2. Pengulangan Bersarang ( NESTED LOOP) 🔂🐦‍

Atau sering disebut pengulangan do-while. Mirip dengan while, namun kondisi pengulangan diperiksa setelah eksekusi blok, sehingga blok kode akan dijalankan setidaknya satu kali.



#🤸🏻‍♀️ EKSPLORASI PRIBADI DI KELAS ✨

In [ ]:
#list
my_list = [1,2,3]
my_list[0]=10
print(my_list)

[10, 2, 3]


In [ ]:
#perulangan for loop
var_list=[1,2,3,4,5,6,7,8,9,10]
for i in var_list:
  print(i)

1
2
3
4
5
6
7
8
9
10


In [ ]:
huruf="Sains Data IKOPIN University"
for i in huruf:
  print(i)

S
a
i
n
s
 
D
a
t
a
 
I
K
O
P
I
N
 
U
n
i
v
e
r
s
i
t
y


In [ ]:
for i in range (1,10,2):
  print(i)

1
3
5
7
9


In [ ]:
for i in range (0,10,1):
  print(i)

0
1
2
3
4
5
6
7
8
9


In [24]:
for i in range (1000):
  print("Hello Word")

Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word
Hello Word

In [ ]:
for i in range (2,5,1):
  print(i)

2
3
4


In [ ]:
for i in range (7,2,-1):
  print(i)

7
6
5
4
3


In [ ]:
a=int(input())
b=int(input())

for i in range(a,b+1):
  print(i)

6
9
6
7
8
9


In [ ]:
#while loop
a=int(input())
b=int(input())

i=a
while(i<=b):
    print(i)
    i+=1

1
5
1
2
3
4
5


In [ ]:
counter=1
while counter <=5:
  print(counter)
  counter +=1


1
2
3
4
5


In [ ]:
n=int(input())

for i in range(n):
  for j in range(n):
    print("*",end="")

2
****

In [ ]:
#nilai kelipatan 5
#Input bilangan N
N=int(input("Masukkan N: "))

#Mencari kelipatan 5 terkecil yang lebih besar dari N
i=N+1
while i%5 !=0: #selama i belum menjadi kelipatan 5
  i+=1 #Tambah 1

#Output hasil
print("output:",i)

Masukkan N: 8
output: 10


In [ ]:
#Program segitiga
#Input dari pengguna
N=int(input("Masukkan nilai N: "))

#Cetak pola segitiga bintang
for i in range(1,N+1):
  print('*'*i)

Masukkan nilai N: 5
*
**
***
****
*****


In [ ]:
#Segitiga Kebalik
N=int(input("Masukkan bilangan N: "))
for i in range(N,0,-1):
  for j in range(1,i+1):
    print(j,end="")
  print()

Masukkan bilangan N: 5
12345
1234
123
12
1


In [14]:
#kelipatan 2 dan 3
N = int(input("Masukkan nilai N: "))

# Loop dari 1 hingga N
for x in range(1, N + 1):
    if x % 2 == 0 and x % 3 == 0:
        print("DuaTiga", end=" ")
    elif x % 2 == 0:
        print("Dua", end=" ")
    elif x % 3 == 0:
        print("Tiga", end=" ")
    else:
        print(x, end=" ")
print()

Masukkan nilai N: 10
1 Dua Tiga Dua 5 DuaTiga 7 Dua Tiga Dua 


#Interpretasi coding:
- dari list 1,2,3 urutan list ke 0 diganti menjadi 10
- ⁠untuk list 1,2,3,4,5,6,7,8,9,10 diulang menjadi urutan kebawah
- untuk huruf : Sains Data IKOPIN University diulang menjadi urutan kebawah ⁠
- ⁠dalam range yang dimulai dari 1 dan di akhiri 10 dengan step 2
- ⁠dalam range yang dimulai dari 0 dan di akhiri 10 dengan step 1
- ⁠print Hello Word sampai 1000
- ⁠dalam range yang dimulai dari 2 dan di akhiri 5 dengan step 1
- ⁠dalam range yang dimulai dari 7 dan di akhiri 2 dengan step -1
- ⁠input nilai a dan b yang dimana  a akan menjadi start kemudian b+1 menjadi akhir dengan step 1
- ⁠masukan n jika i=n+1 bukan kelipatan 5 akan dilanjut dengan i+=1 dan muncul output kelipatan 5 yang lebih dari nilai n
- ⁠masukan nilai n dengan range start 1 diakhiri n+1 dengan step 1 printnya *
- ⁠⁠masukan nilai n dengan range start n diakhiri 0 dengan step -1 kemudian start 1 diakhiri i+1 dengan step 0
- membuat loop dari 1 hingga n jika n keliapatan 2 dan 3 maka hasilnya “DuaTiga” jika n hanya keliapatan 2 maka hasilnya “dua” jika n hanya keliapatan 3 hasilnya “tiga” dan jika hasilnya bukan kelipatan 2 atau 3 maka outputnya angka tersebut

# Interpretasi Hasil:
- otputnya menjadi 10,2,3

- outputnya menjadi 0-10

- outputnya menjadi Sains Data IKOPIN University (tetapi ke bawah satu huruf
satu huruf)

- ouputnya menjadi 1,3,5,7,9

- outputnya menjadi 0-9

- outputnya 1000 Hello Word

- outputnya 2,3,4

- outputnya 7,6,5,4,3
jika a=6 dan b=9 outputnya 6,7,8,9
jika a=1 dan b=5 outputnya 1,2,3,4,5
- output kelipatan 5 yang lebih dari nilai n
-outputnya bintang dengan jumlah 1 2 3 4 5
-outputnya 12345 sampai 1
-outputnyajika n keliapatan 2 dan 3 maka hasilnya “DuaTiga” jika n hanya keliapatan 2 maka hasilnya “dua” jika n hanya keliapatan 3 hasilnya “tiga” dan jika hasilnya bukan kelipatan 2 atau 3 maka outputnya angka tersebut

#🧠 LATIHAN SOAL DI KELAS ⏱️

Problem (1) Kelipatan 10 terkecil

In [22]:
# Input bilangan N
N = int(input("Masukkan N: "))

# Mencari kelipatan 10 terkecil yang lebih besar dari N
i = N + 1
while i % 10 != 0:  # selama i belum menjadi kelipatan 10
    i += 1  # Tambah 1

# Output hasil
print("output:", i)

Masukkan N: 5
output: 10


- Interpretasi coding:

	•	Kita mulai dengan kelipatan awal 10.

  •	Menggunakan loop while, kita akan mengalikan angka tersebut dengan 10 sampai nilainya lebih besar dari N.

- Interpretasi Hasil:

  •	jika N=5, hasilnya akan 10 (karena 10 adalah kelipatan terkecil dari 10 yang lebih besar dari 5).

  •	Jika N=1234, maka program akan melanjutkan perhitungan hingga mencapai 10000 (kelipatan 10 terkecil lebih besar dari 1234).


Untuk soal ini, loop while lebih baik digunakan karena kita tidak tahu secara pasti berapa kali kita harus mengalikan dengan 10 hingga mencapai angka yang lebih besar

Problem (2) Segitiga Angka

In [ ]:
N=int(input("Masukkan bilangan N: "))
for i in range(1,N+1,1):
  for j in range(1,i+1):
    print(j,end="")
  print()

Masukkan bilangan N: 5
1
12
123
1234
12345


- Interpretasi coding:

- Loop luar (for i in range(1, N + 1)) mengatur jumlah baris. Setiap kali i bertambah, kita akan mencetak satu baris baru.
- Loop dalam (for j in range(1, i + 1)) mengatur angka yang dicetak pada setiap baris. Pada baris ke-N, kita mencetak angka dari 1 hingga N.
- Fungsi print(j, end=" ") memastikan angka-angka dicetak dalam satu baris tanpa pindah baris, sedangkan print() setelah loop dalam digunakan untuk berpindah ke baris baru setelah satu baris angka selesai dicetak
- Interpretasi Hasil:

Untuk soal ini, loop bersarang (nested loop) adalah pilihan terbaik karena kita perlu mencetak sejumlah angka yang bervariasi di setiap baris (angka 1 hingga N pada baris ke-N).

Perbandingan Loop pada Kedua Soal:

	•	Kelipatan 10 Terkecil: Loop yang lebih cocok digunakan adalah loop while, karena kita tidak tahu berapa kali kita harus mengalikan dengan 10 hingga mencapai angka yang lebih besar dari . Loop ini lebih dinamis dan kondisional.
	•	Segitiga Angka: Nested loop for lebih cocok digunakan karena kita mengetahui jumlah iterasi yang pasti (berdasarkan nilai ), dan tiap iterasi memiliki batas yang jelas (jumlah angka pada setiap baris).

Kesimpulan Karakteristik Loop:

	•	While loop cocok untuk kasus ketika jumlah iterasi tidak diketahui sebelumnya dan bergantung pada kondisi yang terpenuhi.
	•	For loop lebih cocok ketika jumlah iterasi sudah diketahui atau mudah diprediksi dari input.

#🏆 POST TEST 🎯

Problem (1)

In [26]:
# Input nilai N, A, B, dan C
N = int(input("Masukkan nilai N: "))
A = int(input("Masukkan nilai A: "))
B = int(input("Masukkan nilai B: "))
C = int(input("Masukkan nilai C: "))

# Loop dari 1 sampai N
for x in range(1, N + 1):
    output = ""

    # Cek kelipatan A, B, dan C
    if x % A == 0:
        output += "Siap"
    if x % B == 0:
        output += "Bang"
    if x % C == 0:
        output += "Jago"

    # Jika tidak ada yang diprint (bukan kelipatan A, B, atau C), cetak angka
    if output == "":
        output = str(x)

    # Print hasil untuk x
    print(output, end=" ")

Masukkan nilai N: 12
Masukkan nilai A: 2
Masukkan nilai B: 3
Masukkan nilai C: 4
1 Siap Bang SiapJago 5 SiapBang 7 SiapJago Bang Siap 11 SiapBangJago 

In [16]:
# Input nilai N, A, B, dan C
N = int(input("Masukkan nilai N: "))
A = int(input("Masukkan nilai A: "))
B = int(input("Masukkan nilai B: "))
C = int(input("Masukkan nilai C: "))

# Loop dari 1 sampai N
for x in range(1, N + 1):
    output = ""

    # Cek kelipatan A, B, dan C
    if x % A == 0:
        output += "Siap"
    if x % B == 0:
        output += "Bang"
    if x % C == 0:
        output += "Jago"

    # Jika tidak ada yang diprint (bukan kelipatan A, B, atau C), cetak angka
    if output == "":
        output = str(x)

    # Print hasil untuk x
    print(output, end=" ")


Masukkan nilai N: 5
Masukkan nilai A: 1
Masukkan nilai B: 1
Masukkan nilai C: 1
SiapBangJago SiapBangJago SiapBangJago SiapBangJago SiapBangJago 

- Interpretasi coding:
1. Meminta pengguna untuk memasukkan nilai N, A, B, dan C menggunakan fungsi input() dan menyimpan hasilnya dalam variabel yang sesuai.
2. Membuat sebuah loop yang berjalan dari 1 hingga N menggunakan fungsi range() dan menyimpan setiap nilai dalam variabel i.
3. Memeriksa apakah i adalah kelipatan dari A, B, atau C menggunakan operator modulus %. Jika i adalah kelipatan dari salah satu dari mereka, maka mencetak kata yang sesuai ("Siap", "Bang", atau "Jago") menggunakan fungsi print().
4. Jika i bukan kelipatan dari A, B, atau C, maka mencetak nilai i menggunakan fungsi print().

- Interpretasi hasil:
Dengan kode tersebut, kita dapat menghasilkan output yang diinginkan berdasarkan input yang diberikan oleh pengguna.


📌Catatan:
  - Notebook python ini dibuat untuk memudahkan temen-temen belajar ngoding dan pengumpulan tugas.
  - Silakan download notebook ini lalu upload di Google Collab milik kalian agar format `pertemuan`, `topik pertemuan`,
`tanggal`, `deskripsi pembahasan materi`, `baris coding` dan `interpretasi` bisa disesuaikan dengan kebutuhan teman-teman.

  - Interpretasi coding dan hasil dimaksudkan agar teman-teman semakin memahami apa yang dipelajari.

  - Deskripsi pembahasan materi dimaksudkan agar teman-teman semakin memahami apa fokus yang dipelajari.

  - Tulisan di setiap sebelum baris code disebut `markdown`. Ada beberapa jenis simbol yang memudahkan kepenulisan `markdown` kita, yaitu seperti : untuk mengatur ukuran tulisan biasanya tulisan pada header menggunakan ini`#`, `##`, `###`, untuk membuat pointer menggunakan ini `-`, untuk membuat tulisan ter-highlight menggunakan ini *` `* , untuk short-cut membuat tulisan jadi **bold** gunakan ini `** **`, dan untuk *tulisan miring* gunakan ini `* *` .

  - Semangat belajar! 🌟✨

©️ LABORATORIUM SAINS DATA 2024